### DISCUSSION TITLE
wifi treaming example issue with my own wifi network

### DISCUSSION THREAD

**Discussion by ClementSrs** [bitcraze/discussions#1055] [Q&A]
Hi ! 
With the menuconfig a I'm able to connect the CF with my wifi network, and now I want to recover the images provide by the ai-deck .
I assume that the aideck IP is 192.168.0.14.

$ python3 opencv-viewer.py -n 192.168.0.14 --save
Connecting to socket on 192.168.0.14:5000...
Socket connected

After this, there is no images provide by the CF. There is another thing to change in the CF's firmware ? I think my computer doesn't search the information in the good place.

Thank you.
Clement

**Comment by ClementSrs** [comment]
I already re-flash the esp32-firmware with the crazyradio (https://github.com/bitcraze/aideck-esp-firmware/tree/main)

**Comment by knmcguire** [comment]
Hi! Have you also reflashed a rebuild wifi example on the gap8 as well according to these instructions? https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/simple-examples/wifi-streamer/. '

The example wifi example from the [getting started tutorial ](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/) is 'deaf' to any instructions of the crazyflie firmware.

**Reply by ClementSrs** [reply]
Yes I did all of this. 
I found the crazyflie on my local network, I established the connexion ( verification with the client ). 
The socket is connected but I images didn't appear.

**Reply by knmcguire** [reply]
Hmm... I'm wondering then if the camera actually starts properly. You can verify that by adding cpx console messages if it goes past the camera initialization in the gap8 code.

**Reply by ClementSrs** [reply]
How can I do that ? 
Do you ever try to recover images in the same way ? (specific wifi network).

**Reply by knmcguire** [reply]
Yes, we have been busy with release testing and letting the crazyflie connect to a wifi access point has been one of the tests, which worked fine for us. So there is something else going on in your situation that you'll need to debug.

I'm talking about the cpxtoconsole function like this one: https://github.com/bitcraze/aideck-gap8-examples/blob/c73d9ece5f28bef0e5f31c39c4fc9b5d77bf851d/examples/other/wifi-img-streamer/wifi-img-streamer.c#L208. You can add similar messages to see if the code goes through any of the initialization points in one piece before actually sending the images.

**Reply by ClementSrs** [reply]
Do the actual opencv_viewer.py is compatible with this kind of connexion ? I didn't find someone with the same issue... If you know someone who is revovering aideck's images via a wifi network please give me the contact. I'm stuck for few days. Thank you

**Reply by knmcguire** [reply]
yes it should be compatible. I'll start  another thread with some of the things that I see.

**Comment by knmcguire** [comment]
So I can't try it out on my own wifi network as my home wifi network has settings that blocks the connection (perhaps you have the same) and I don't have time now to fix that later. At least when you set the crazyflie as an access point itself you should see this in the cfclient console:

```
SYS: ----------------------------
SYS: Crazyflie 2.1 is up and running!
SYS: Build 0:745fb09397e3 (2023.11) CLEAN
SYS: I am 0x20393743484850170036004C and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 1 deck(s) found
DECK_CORE: Calling INIT on driver bcAI for deck 0
CPX-EXT-ROUTER: Waiting for CPX External router initialization
CPX-EXT-ROUTER: CPX External router initialized, CPX_VERSION: 0
AIDECK: AI-deck will become access point
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Complementary (1) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2023.11
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
DECK_CORE: Deck 0 test [OK].
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 14600 bytes
CPX: ESP32: I (800) SYS: Initialized
CPX: ESP32: I (800) DISCOVERY: Hostname is aideck-E0CB08
CPX: ESP32: I (800) DISCOVERY: MAC (STA) is 9c:9c:1f:e0:cb:8
CPX: GAP8: -- WiFi image streamer example --
CPX: ESP32: I (2730) WIFI: wifi_init_softap finished
STAB: Starting stabilizer loop
SUP: Can not fly
SUP: Ready to fly
CPX: GAP8: Starting camera task...
CPX: ESP32: I (8780) WIFI: station:70:a6:cc:0f:72:d8 join, AID=1
CPX: ESP32: I (49930) WIFI: Connection accepted
CPX: ESP32: I (49930) WIFI: Client connected
CPX: CPX connected
CPX: GAP8: Wifi client connection status: 1
CPX: GAP8: capture=74ms, encoding=0 ms (79056 bytes), transfer=7 ms
CPX: GAP8: capture=74ms, encoding=0 ms (79056 bytes), transfer=63 ms
CPX: GAP8: capture=75ms, encoding=0 ms (79056 bytes), transfer=60 ms
CPX: GAP8: capture=74ms, encoding=0 ms (79056 bytes), transfer=61 ms
....
```

And in the terminal where you started the viewer:
```
python examples/other/wifi-img-streamer/opencv-viewer.py 
Connecting to socket on 192.168.4.1:5000...
Socket connected
0.10144996643066406
9.857075711143281
0.11588525772094727
8.629225318789116
0.12161890665690105
...
```
And images showing as seperate images.
This should, for the exception of a few messages in the console related to connecting to a remote wifi network, be pretty much the same for you if everything is working correctly.

Further followup question for you is:
- Does the example work with the aideck as access point?
- Is the wifi network you'd like to connect to a lab or home network that you have access to the settings?
- On what OS are you running the opencv viewer from?

**Reply by ClementSrs** [reply]
Hi ! 
I have all the command in my cdclient console without the last ones :

CPX: GAP8: Wifi client connection status: 1
CPX: GAP8: capture=74ms, encoding=0 ms (79056 bytes), transfer=7 ms
CPX: GAP8: capture=74ms, encoding=0 ms (79056 bytes), transfer=63 ms
CPX: GAP8: capture=75ms, encoding=0 ms (79056 bytes), transfer=60 ms
CPX: GAP8: capture=74ms, encoding=0 ms (79056 bytes), transfer=61 ms

What am I missing ?

Thank you

**Reply by knmcguire** [reply]
Could you answer also my last 3 questions please? Then it will be perhaps more clear to us what is going on

**Comment by Serebrianka** [comment]
What worked for us was to : 

- Flash the esp32 firmware with JTAG following this : https://github.com/bitcraze/aideck-esp-firmware
- Flash the crazyflie firmware with a Kbuild customized firmware : expansion decks -> ai deck -> wifi at startup -> connect to a wifi network -> ceditentials for access point (SSID and Password). Then we saved this config with default name. After we built the firmware using "make -j 12", we turned on our Crazyflie in bootloader mode (pressing the power button for 3 seconds) and flash over the air using "make cload"
- Flash the wifi streamer example from the github : https://github.com/bitcraze/aideck-gap8-examples and following the docs : https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/simple-examples/wifi-streamer/
- After we runed the example with our Ai-Deck IP

I am pretty sure you have done all this correctly, but this worked for us.
